cg/store/crud/read.py

```diff
@@ -456,6 +444,49 @@ def get_sample_lane_sequencing_metrics_by_flow_cell_name(
             flow_cell_name=flow_cell_name,
         ).all()
 
+    def get_illumina_metrics_entry_by_device_sample_and_lane(
+        self, run_id: str, sample_internal_id: str, lane: int
+    ) -> IlluminaSampleSequencingMetrics:
+        """Get metrics entry by sequencing device internal id, sample internal id and lane."""
+        filtered_samples: Subquery = apply_sample_filter(
+            samples=self._get_query(table=Sample),
+            filter_functions=[SampleFilter.BY_INTERNAL_ID],
+            internal_id=sample_internal_id,
+        ).subquery()
+        filtered_metrics: Subquery = apply_illumina_metrics_filter(
+            metrics=self._get_query(table=IlluminaSampleSequencingMetrics),
+            filter_functions=[IlluminaMetricsFilter.BY_LANE],
+            lane=lane,
+        ).subquery()
+        filtered_flow_cells: Subquery = apply_illumina_flow_cell_filters(
+            flow_cells=self._get_query(table=IlluminaFlowCell),
+            filter_functions=[IlluminaFlowCellFilter.BY_INTERNAL_ID],
+            internal_id=run_id,
+        ).subquery()
```

Comment:
Why do we need subqueries here? is it not simpler to just join and filter? I would prefer that since this looks quite complex in comparison

Reply:
It will join the whole tables at once which can become expensive once the tables grow (i.e. joining all samples with all sample sequencing metrics).